File: JAVA/Clothing.java
class Clothing extends Product {    //class clothing yang medapatkan atribut turunan dari class product
    String size;    //atribut dari kelas clothing
    String material;
    char gender;

    Clothing(){ //konstruktor kosong
    }

    Clothing (String size, String material, char gender){   //konstruktor dengan parameter
        this.size = size;
        this.material = material;
        this.gender = gender;
    }

    //get dan set
    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getMaterial() {
        return material;
    }

    public void setMaterial(String material) {
        this.material = material;
    }

    public char getGender() {
        return gender;
    }

    public void setGender(char gender) {
        this.gender = gender;
    }
}

File: CPP/PROGRAM/Clothing.cpp
#include <iostream>
#include <string>
#include "Product.cpp"  //menginclude nama file orangtua untuk mewariskan object dari class orang tua

//using standard namespace
using namespace std;

//class clothing yang mewarisi dari kelas product
class Clothing : public Product{
    private:
        string size;
        string material;
        char gender;

    public:
        Clothing(){
        }

        //getter dan setter
        void set_size(string size){
            this->size = size;
        }

        string get_size(){
            return this->size;
        }

        void set_material(string material){
            this->material = material;
        }

        string get_material(){
            return this->material;
        }

        void set_gender(char gender){
            this->gender = gender;
        }

        char get_gender(){
            return this->gender;
        }

        ~Clothing(){
        }
};